web/olga/analytics/views.py

```diff
@@ -199,7 +213,83 @@ def extend_stats_to_enthusiast(self, received_data, stats, edx_installation_obje
         edx_installation_object.platform_url = enthusiast_edx_installation['platform_url']
         edx_installation_object.save()
 
-    def create_instance_data(self, received_data, access_token):
+    @atomic
+    def process_instance_datas(self, received_data, access_token):
+        today_date = datetime.datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
+        edx_installation_object = EdxInstallation.objects.get(access_token=access_token)
+        today_stats = self.get_today_stats(received_data, edx_installation_object)
+        dates = self.get_all_stats_by_dates(received_data, access_token)
+        self.add_today_to_dates(today_date, today_stats, dates)
+
+        for date in sorted(dates.keys()):
+            statistics = dates[date]
+            self.create_instance_data(statistics, edx_installation_object, date)
+
+    def add_today_to_dates(self, today_date, today_stats, dates):
+        """
+        Add today statistics data to the dates dictionary for the further pass to the create_instance_data.
+        """
+        if today_date not in dates:
+            dates[today_date] = today_stats
+            return
+
+        today_stats['registered_students'] = dates[today_date]['registered_students']
+        today_stats['enthusiastic_students'] = dates[today_date]['enthusiastic_students']
+        today_stats['generated_certificates'] = dates[today_date]['generated_certificates']
+        dates[today_date] = today_stats
+
+    def get_all_stats_by_dates(self, received_data, access_token): 
+        """
+        Return a dictionary of dates and stats (as the keys and the values respectively) to create the object.
+        """
+        dates = {}
+        data_template = {
```

Comment:
Only first three parameters are updated from the `received_data` in this function other could be removed.

Reply:
done